include/graphblas/bsp1d/blas3.hpp

```diff
@@ -205,6 +205,86 @@ namespace grb {
 		return internal::checkGlobalErrorStateOrClear( C, ret );
 	}
 
+	template<
+		Descriptor descr = descriptors::no_operation,
+		class SelectionOperator,
+		typename Tin,
+		typename RITin, typename CITin, typename NITin,
+		typename Tout,
+		typename RITout, typename CITout, typename NITout
+	>
+	RC select(
+		Matrix< Tout, BSP1D, RITout, CITout, NITout >& out,
+		const Matrix< Tin, BSP1D, RITin, CITin, NITin >& in,
+		const SelectionOperator &op,
+		const Phase& phase = EXECUTE,
+		const typename std::enable_if<
+				!is_object< Tin >::value &&
+				!is_object< Tout >::value &&
+				is_matrix_selection_operator< SelectionOperator >::value
+		>::type * const = nullptr
+	) {
+		assert( phase != TRY );
+
+		RC ret = internal::select_generic< descr >(
+			internal::getLocal( out ),
+			internal::getLocal( in ),
+			in.getLocalCoordinatesOffset(),
+			op,
+			phase
+		);
+
+		if( phase == RESIZE ) {
+			if( collectives<>::allreduce( ret, operators::any_or< RC >() ) != SUCCESS ) {
+				return PANIC;
+			} else {
+				return SUCCESS;
+			}
+		}
+		assert( phase == EXECUTE );
+		return internal::checkGlobalErrorStateOrClear( out, ret );
+	}
```

Comment:
Passing in a generic distribution / how to get from local to global has the added benefit that it will work with any other distributed-memory backend (which may make other distribution choices)